### PR TITLE
Add new UI and API for Toggling Endpoint Access

### PR DESCRIPTION
**Pull request by MongoCaleb**
### Jira

https://jira.mongodb.org/browse/DOCSP-33237

### Staged Changes

- https://preview-mongodbmongocaleb.gatsbyjs.io/atlas-app-services/docsp-33237/security/private-endpoints/
https://preview-mongodbmongocaleb.gatsbyjs.io/atlas-app-services/docsp-33237/admin/api/v3/#tag/security/operation/allNonVpcClientRequests

### CODE REVIEW

File: source/openapi-admin-v3.yaml [hunk 1]
```diff
@@ -2596,6 +2596,34 @@ paths:
                 type: array
                 items:
                   $ref: "#/components/schemas/PrivateEndpointServiceInfo"
+
+  "/groups/{groupId}/apps/{appId}/security/allow_non_vpc_client_requests":
+    parameters:
+      - "$ref": "#/components/parameters/GroupId"
+      - "$ref": "#/components/parameters/AppId"
+    post:
+      operationId: allNonVpcClientRequests
```
Comment (nlarew) [inline source/openapi-admin-v3.yaml:2605]:
```suggestion
      operationId: allowNonVpcClientRequests
```

File: source/security/private-endpoints.txt [hunk 2]
```diff
@@ -35,8 +35,12 @@ Private Endpoint Limitations
   connections over the MongoDB wire protocol <wire-protocol>`. If one is
   enabled, you cannot enable the other.
 
-- Private Endpoints only handle internal requests. Any outbound requests, such as 
-  an an App Services trigger calling an external API, go over the public internet. 
+- By default, private Endpoints only handle internal requests. You can  
+  :ref:`change access restrictions <change_access_restrictions>` from all 
+  requests outside your PrivateLink connection.
```
Comment (nlarew) [inline source/security/private-endpoints.txt:40]:
```suggestion
- By default, private Endpoints only handle requests that originate from within your VPC.
  You can configure your App to also accept requests from the public internet. To learn how,
  see :ref:`Change Access Restrictions <change_access_restrictions>`.
```

File: source/security/private-endpoints.txt [hunk 3]
```diff
@@ -139,21 +140,33 @@ connection using either the App Services UI or the Admin API.
...
+                    --vpc-endpoint-id <your-vpc-endpoint-id> \
+                    --private-dns-enabled
+            
+            #. If this is the first Private Endpoint you are creating, you will 
+               see a toggle you can use to either restrict or open all access from 
+               requests outside your PrivateLink connection. By default, App 
+               Services blocks all requests coming from outside your PrivateLink 
+               connection. This setting applies to this endpoint and **all 
+               additional endpoints you create**. Refer to :ref:`change access 
+               restrictions <change_access_restrictions>` for more information.
```
Comment (nlarew) [inline source/security/private-endpoints.txt:166]: Can we add an equivalent step to the Admin API procedure step 2? i.e. a new field in the request body (if possible)

**Review by nlarew — APPROVED** [review]
LGTM % one possible addition to the Admin API procedure

File: source/security/private-endpoints.txt (hunk 3 above)
Comment (MongoCaleb) [reply source/security/private-endpoints.txt:166]: There is no new field in the request body, alas. It can only be set via the new endpoint.

**Pull request merged**